R/sprintf_linter.R

```diff
@@ -1,7 +1,25 @@
 #' Require correct `sprintf()` calls
 #'
-#' Check for an inconsistent number of arguments or arguments with incompatible types (for literal arguments) in
-#' `sprintf()` calls.
+#' Check for an inconsistent number of arguments or arguments with incompatible types
+#' (for literal arguments) in `sprintf()` calls.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "sprintf('hello %1$s %1$s %3$d', x, y, z)",
```

Comment:
I find it odd to be using the 1$ redirections, those are rarely seen in practice IME. Let's just stick with plain templates which should be easier to read